includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,470 @@
+<?php
+/**
+ * Class Sanitization
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\AMP;
+
+use AMP_Allowed_Tags_Generated;
+use AMP_Content_Sanitizer;
+use AMP_DOM_Utils;
+use AMP_Layout_Sanitizer;
+use AMP_Script_Sanitizer;
+use AMP_Style_Sanitizer;
+use AMP_Tag_And_Attribute_Sanitizer;
+use AmpProject\Amp;
+use AmpProject\Attribute;
+use AmpProject\Dom\Document;
+use AmpProject\Extension;
+use AmpProject\Tag;
+use DOMElement;
+
+/**
+ * Sanitization class.
+ *
+ * @since 1.0.0
+ */
+class Sanitization {
+	/**
+	 * Sanitizes a document.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param Document $document Document instance.
+	 *
+	 * @return void
+	 */
+	public function sanitize_document( Document $document ) {
+		$sanitizers = $this->get_sanitizers();
+		/**
+		 * Class name.
+		 *
+		 * @var string $sanitizer_class
+		 */
+		foreach ( $sanitizers as $sanitizer_class => $args ) {
+			/**
+			 * The add_buffering_hooks callable.
+			 *
+			 * @var callable $callable
+			 */
+			$callable = [ $sanitizer_class, 'add_buffering_hooks' ];
+			if ( is_callable( $callable ) ) {
+				call_user_func( $callable, $args );
+			}
+		}
```

Comment:
The `add_buffering_hooks` section can be omitted as this is intended to be invoked before the template starts rendering. Since this is invoked after the document has already been rendered, it will have no purpose.

Reply:
```suggestion
```